src/relayclient/types/RelayInfo.ts

```diff
@@ -1,9 +1,14 @@
 import PingResponse from '../../common/PingResponse'
-import RelayRegisteredEventInfo from './RelayRegisteredEventInfo'
+import { RelayInfoUrl, RelayRegisteredEventInfo } from './RelayRegisteredEventInfo'
 
 // Well, I still don't like it
 // Some info is known from the event, some from ping
-export default interface RelayInfo {
+export interface PartialRelayInfo {
```

Comment:
I really don't understand your nested structure. why relayInfo has a relayInfo member?
is inheriting (and then checking that members are missing) is any better than optional fields?

Reply:
Optional fields remain optional all the way. So if you have
``` let a = getWithOptional()
a = fillMissingFields(a)

console.log(a.foo) 
                     ^ ... error: foo is 'string | undefined'
```
This way I convert from a class with optional fields to one with all fields assigned.